nagios_exporter.py

```diff
@@ -235,16 +433,23 @@ def collect_metrics(args, lines):
         session = LiveStatus(sock)
         lines.extend(get_status(session))
 
-    with contextlib.closing(connect(args.path)) as sock:
-        # TODO: support filtering specific services or hosts.
-        session = LiveStatus(sock)
-        lines.extend(get_services(session))
+    if args.whitelist or args.all_metrics or args.dump_metrics:
+        with contextlib.closing(connect(args.path)) as sock:
+            session = LiveStatus(sock)
+            services = get_services(
+                session, args.use_perf_data, args.data_names)
+
+        if args.whitelist:
+            for whitelist in args.whitelist:
+                lines.extend(filter(lambda x: whitelist in x, services))
```

Comment:
This could potentially put items from `service` in `lines` multiple times. Is this okay?